lib/SharedCUEParser/SharedCUEParser.cpp

```diff
@@ -0,0 +1,97 @@
+    
+#include "scp/SharedCUEParser.h"
+#include <string.h>
+#include <SdFat.h>
+
+extern SdFs SD;
+char SharedCUEParser::_current_file_loaded[CUE_MAX_FILENAME + 1] = {0};
+char SharedCUEParser::_shared_cuesheet[MAX_SHARED_CUE_SHEET_SIZE];
+
+static void write_default_cuesheet(char * cue_sheet)
+{
+    strcpy(cue_sheet, R"(
+    FILE "" BINARY
+    TRACK 01 MODE1/2048
+    INDEX 01 00:00:00
+    )");
+}
+
+SharedCUEParser::SharedCUEParser()
+{
+    _cue_filepath[0] = '\0';
+    m_cue_sheet = _shared_cuesheet;
+    if (_shared_cuesheet[0] == '\0')
+    {
+        write_default_cuesheet(_shared_cuesheet);
+    }
+    restart();
+}
+
+SharedCUEParser::SharedCUEParser(char* path)
```

Comment:
Using file paths as the key here seems to needlessly complicate logic in the rest of the application, which already has a `FsFile` available.

How about this:

* SharedCUEParser constructor takes FsFile and keeps it open in a member variable `_cue_file`.
* `update_file()` compares `_cue_file.firstCluster()` against `_shared_cue_first_cluster`.

Idea being that two different files can never start at the same cluster.

This needs documenting that the `FsFile` given to `SharedCUEParser` should not be closed by the caller. It could be cleanest to add a `~SharedCUEParser` destructor that closes the file, but for read-only files closing does not really do anything.